ext/ffi/lib.rs

```diff
@@ -289,6 +318,7 @@ union NativeValue {
   f32_value: f32,
   f64_value: f64,
   pointer: *const u8,
+  structure_value: *mut [u8],
```

Comment:
This is no longer really used, right? Remove it and the few places that still use it.

Alternatively, if the `*mut` part is necessary to avoid some parts then use the `structure_value` variant everywhere. Currently the struct code uses `pointer` and `structure_value` interchangeably.

Reply:
struct_value has a distinct use now, no longer reusing pointer